### DISCUSSION TITLE
Apache configuration issue, or maybye an another issue ?

### DISCUSSION THREAD

**Discussion by feydreva** [sympa-community/sympa#1670] [Q&A] [answered]
Hello, 

I am running Sympa on a ubuntu server, running apache hosting a wikimedia and sympa, with let's encrypt
Wiki is on domain wiki.domain.xxx
Sympa in on domain listes.domain.xxx

no issue with wiki, was there before, and working fine.

I finished the installation of sympa following instruction on <https://www.sympa.community/manual/install/> 
(it was painfull, installation instructions are not clear at all :/ but hats not the point for today)

I installed sympa with systemd socket.
everything looks good, but when I access the Sympa page at listes.domain.xxx 
I end up at the default Apache2 Ubuntu default page : it works.

Apache logs : 

> listes.domain.xxx:443 xxx.xxx.xxx.xxx - - [28/May/2023:02:21:59 +0200] "GET / HTTP/1.1" 200 8780 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/113.0"
listes.domain.xxx:443 xxx.xxx.xxx.xxx - - [28/May/2023:02:21:59 +0200] "GET /icons/ubuntu-logo.png HTTP/1.1" 200 3667 "https://listes.domain.xxx/" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/113.0"
listes.domain.xxx:443 xxx.xxx.xxx.xxx - - [28/May/2023:02:21:59 +0200] "GET /favicon.ico HTTP/1.1" 404 520 "https://listes.domain.xxx/" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/113.0"

apache2 listes.domain.xxx.conf

```
<IfModule mod_ssl.c>
<VirtualHost *:443>
    ServerName listes.domain.xxx
    ServerAdmin admin@domain.xxx

    LoadModule alias_module modules/mod_alias.so
    LoadModule proxy_module modules/mod_proxy.so
    LoadModule proxy_fcgi_module modules/mod_proxy_fcgi.so

    <Location /sympa>
        SetHandler "proxy:unix:/run/sympa/wwsympa.socket|fcgi:/localhost/"
        #Require local
        Require all granted
    </Location>

# Lines below are needed for 6.2.28 or later.
    <Location "/static-sympa/css">
        Require all granted
    </Location>
    Alias /static-sympa/css /var/lib/sympa/css

# Lines below are needed for 6.2.28 or later.
    <Location "/static-sympa/pictures">
        Require all granted
    </Location>
    Alias /static-sympa/pictures /var/lib/sympa/pictures

    <Location /static-sympa>
        Require all granted
    </Location>
    Alias /static-sympa /usr/share/sympa/static_content


SSLCertificateFile /etc/letsencrypt/live/wiki.domain.xxx-0001/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/wiki.domain.xxx-0001/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf
</VirtualHost>
</IfModule>
```

wwsympa socket looks to be started ok : 

> root@web:/var/log/apache2# systemctl status wwsympa.socket
● wwsympa.socket - Sympa web interface socket
     Loaded: loaded (/lib/systemd/system/wwsympa.socket; disabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/wwsympa.socket.d
             └─override.conf
     Active: active (listening) since Sun 2023-05-28 01:59:54 CEST; 29min ago
   Triggers: ● wwsympa.service
     Listen: /run/sympa/wwsympa.socket (Stream)
      Tasks: 0 (limit: 2280)
     Memory: 20.0K
     CGroup: /system.slice/wwsympa.socket
May 28 01:59:54 web systemd[1]: Starting Sympa web interface socket.
May 28 01:59:54 web systemd[1]: Listening on Sympa web interface socket.

wwsympa.service looks OK (Maybe somehwere in the documentation, you should mention multiwatch needs to be installed)
>root@web:/var/log/apache2# systemctl status wwsympa.service
● wwsympa.service - Sympa web interface FastCGI backend
     Loaded: loaded (/lib/systemd/system/wwsympa.service; disabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/wwsympa.service.d
             └─override.conf
     Active: active (running) since Sun 2023-05-28 02:48:08 CEST; 23min ago
TriggeredBy: ● wwsympa.socket
   Main PID: 1798320 (multiwatch)
      Tasks: 6 (limit: 2280)
     Memory: 503.5M
     CGroup: /system.slice/wwsympa.service
             ├─1798320 /usr/bin/multiwatch -f 5 -- /usr/lib/cgi-bin/sympa/wwsympa.fcgi
             ├─1798321 /usr/bin/perl /usr/lib/cgi-bin/sympa/wwsympa.fcgi
             ├─1798322 /usr/bin/perl /usr/lib/cgi-bin/sympa/wwsympa.fcgi
             ├─1798323 /usr/bin/perl /usr/lib/cgi-bin/sympa/wwsympa.fcgi
             ├─1798324 /usr/bin/perl /usr/lib/cgi-bin/sympa/wwsympa.fcgi
             └─1798325 /usr/bin/perl /usr/lib/cgi-bin/sympa/wwsympa.fcgi
May 28 02:48:08 web systemd[1]: Started Sympa web interface FastCGI backend.
May 28 02:48:14 web wwsympa[1798325]: info main:: WWSympa started, process 1798325
May 28 02:48:14 web wwsympa[1798323]: info main:: WWSympa started, process 1798323
May 28 02:48:14 web wwsympa[1798324]: info main:: WWSympa started, process 1798324
May 28 02:48:14 web wwsympa[1798321]: info main:: WWSympa started, process 1798321
May 28 02:48:14 web wwsympa[1798322]: info main:: WWSympa started, process 1798322

   
   I don't know where I am missing something.
   I don't see any error in any logs...
   I search the web, the documentation, but could not find any help.
   
   So thank you for helping me :)

**Comment by ikedas** [comment, marked as answer]
Hi @feydreva ,

Could you please access to `https://listes.domain.xxx/sympa` (this location should be set with `wwsympa_url` parameter in `sympa.conf` or `robot.conf` file) and check Apache's log?

**Comment by feydreva** [comment]
Hello, 

I have set up wwsympa_url to https://listes.domain.xxx/sympa (previoulsy https://listes.domain.xxx)
but in both case I have the following error on apache logs : 
>[Sun May 28 14:52:53.747764 2023] [proxy:error] [pid 1799068] (13)Permission denied: AH02454: FCGI: attempt to connect to Unix domain socket /run/sympa/wwsympa.socket (*) failed
[Sun May 28 14:52:53.750769 2023] [proxy_fcgi:error] [pid 1799068] [client xx.xx.xx.xx:10922] AH01079: failed to make connection to backend: httpd-UDS


I ran the commad sympa-wizard --check, as proposed on other post of people having some issue.  but all seem good.
```
root@web:/usr/share/sympa/bin# sympa_wizard --check
##############################################################################
# This process will help you install all Perl (CPAN) modules required by Sympa
# software.
# Sympa requires from 50 to 65 additional Perl modules to run properly. 
# The whole installation process should take around 15 minutes.
# You'll first have to configure the CPAN shell itself and select your
# favourite CPAN server.
# Note that you might prefer to install the required Perl modules using your
# favourite DEB/RPM mechanism.
# Feel free to interrupt the process if needed ; you can restart it safely
# afterward.
##############################################################################
Press the Enter key to continue...



Checking for PERL version:

Your version of perl is OK (5.030000  >= 5.8.1)

Checking for REQUIRED modules:

perl module                     STATUS
-----------                     ------
Archive::Zip                    OK (1.67   >= 1.05)
CGI                             OK (4.46   >= 3.51)
CGI::Fast                       OK (2.15   >= 1.08)
Class::Singleton                OK (1.5    >= 1.03)
DBD::mysql                      OK (4.050  >= 4.008)
DBI                             OK (1.643  >= 1.48)
DateTime::Format::Mail          OK (0.403  >= 0.28)
DateTime::TimeZone              OK (2.38   >= 0.59)
Digest::MD5                     OK (2.55   >= 2.00)
Encode                          OK (3.02   >= 1.0)
FCGI                            OK (0.79   >= 0.67)
File::Copy::Recursive           OK (0.45   >= 0.36)
File::NFSLock                   OK (1.29   >= 1.0)
File::Path                      OK (2.16   >= 2.08)
HTML::FormatText                OK (2.12   >= 1.0)
HTML::StripScripts::Parser      OK (1.03   >= 1.03)
HTML::TreeBuilder               OK (5.07   >= 1.0)
IO::File                        OK (1.40   >= 1.10)
IO::Scalar                      OK (2.111  >= 1.0)
LWP::UserAgent                  OK (6.43   >= 1.0)
List::Util::XS                  OK (1.5    >= 1.20)
Locale::Messages                OK (1.26   >= 1.20)
MHonArc::UTF8                   OK (2.6.19+ >= 1.0)
MIME::Base64                    OK (3.15   >= 3.03)
MIME::Charset                   OK (1.012.2 >= 1.011.3)
MIME::EncWords                  OK (1.014.3 >= 1.014)
MIME::Lite::HTML                OK (1.24   >= 1.23)
MIME::Tools                     OK (5.509  >= 5.423)
Mail::Address                   OK (2.21   >= 1.70)
Net::CIDR                       OK (0.20   >= 0.16)
Sys::Syslog                     OK (0.35   >= 0.03)
Template                        OK (2.27   >= 2.21)
Term::ProgressBar               OK (2.22   >= 2.09)
Text::LineFold                  OK (2018.012 >= 2011.05)
Time::HiRes                     OK (1.976  >= 1.29)
URI::Escape                     OK (3.31   >= 3.28)
XML::LibXML                     OK (2.0134 >= 1.70)

Checking for OPTIONAL modules:

perl module                     STATUS
-----------                     ------
AuthCAS                         was not found on this system.
Setting FTP Passive mode
-> Usage of this module: CAS Single Sign-On client libraries. Required if you
   configure Sympa to delegate web authentication to a CAS server.
-> Install module AuthCAS ? [n] 
Clone                           OK (0.43   >= 0.31)
Crypt::CipherSaber              OK (1.01   >= 0.50)
Crypt::Eksblowfish              OK (0.009  >= 0.009)
Crypt::OpenSSL::X509            OK (1.813  >= 1.800.1)
Crypt::SMIME                    OK (0.25   >= 0.15)
DBD::CSV                        OK (0.53   >= 0.22)
DBD::ODBC                       was not found on this system.
-> Usage of this module: ODBC database driver, required if you connect to a
   database via ODBC.
-> Install module DBD::ODBC ? [n] 
DBD::Oracle                     was not found on this system.
-> Usage of this module: Oracle database driver, required if you connect to a
   Oracle database.
-> Install module DBD::Oracle ? [n] 
DBD::Pg                         OK (3.10.4 >= 2.00)
DBD::SQLite                     OK (1.64   >= 1.31)
Data::Password                  was not found on this system.
-> Usage of this module: Used for configureable hardening of passwords via the
   password_validation sympa.conf directive.
-> Install module Data::Password ? [n] 
Encode::Locale                  OK (1.05   >= 1.02)
IO::Socket::SSL                 OK (2.067  >= 0.90)
Mail::DKIM::ARC::Signer         
OK (0.58   >= 0.51)
Mail::DKIM::Verifier            OK (0.58   >= 0.37)
Net::DNS                        OK (1.22   >= 0.65)
Net::LDAP                       OK (0.66   >= 0.40)
Net::SMTP                       OK (3.11   >= 1.0)
SOAP::Lite                      OK (1.27   >= 0.712)
Unicode::Normalize              OK (1.26   >= 0.23)
******* NOTE *******
You can retrieve all theses modules from any CPAN server
(for example ftp://ftp.pasteur.fr/pub/computing/CPAN/CPAN.html)
```


Thnak you for you help

**Comment by feydreva** [comment]
Hello, 

after looking at documentation, looking at similar error on internet
this modification did the trick : 

>/etc/systemd/system/wwsympa.socket.d/override.conf:
[Socket]
SocketUser=www-data

documentation says 
>SocketUser=Apache

I did that during initial set up, but service would not start. So I removed it.
Debian and Ubuntu based server use www-data for web server user. maybe a note should be added in documentation.

Thank you for your help.

What should I do now to have my listes site at listes.domain.xxx and not lites.domain.xxx/sympa ?
thank you
fey

**Comment by ikedas** [comment]
Hi @feydreva ,

Debian/Ubuntu package provides Systemd units as you wish.  For example:
https://salsa.debian.org/sympa-team/sympa/-/blob/master/debian/sympa.wwsympa.socket

If you do not use the package provided by the OS, you may need to reread the description in our manual for your environment so that you perform the installation appropriately.
